langserver/references.go

```diff
@@ -85,6 +85,23 @@ func (h *LangHandler) handleTextDocumentReferences(ctx context.Context, conn JSO
 		return PathHasPrefix(path, h.init.RootImportPath)
 	}
 
+	var (
+		// locsC receives the final collected references via
+		// refStreamAndCollect.
+		locsC = make(chan []lsp.Location)
+
+		// refs is a stream of raw references found findReferences.
+		refs = make(chan *ast.Ident)
+
+		// findRefErr is non-nil if findReferences fails.
+		findRefErr error
+	)
+
+	go func() {
+		locsC <- refStreamAndCollect(ctx, conn, req, fset, refs)
```

Comment:
this really should never panic, we control all the code in it, including libraries it calls. I'd rather it take down the server so we notice it.

Other places we have panic handlers in goroutines is due to using `go/types`/etc libraries on all the code out in the wild.